### ISSUE TITLE
[FEATURE]: Staff position customization features

### ISSUE THREAD

**Issue by 22029144d** [COMP3122-2324-Project/Simpsyber#5]
Allowing players to choose staff for different positions at the start of the game or at certain points throughout can provide a more personalized and strategic experience. At the beginning of the game, players are presented with a list of the number of vacancies for each role. Players can then choose number of position based on their strategic priorities.

![image](https://github.com/COMP3122-2324-Project/Simpsyber/assets/157147838/b17af0e9-4e59-4bbc-95d1-ccb5171d9a69)

**Comment by 22029144d** [comment]
Allowing players to choose staff for different positions at the start of the game or at certain points throughout can provide a more personalized and strategic experience. At the beginning of the game, players are presented with a list of the number of vacancies for each role. Players can then choose number of position based on their project requirements and strategic priorities.

![image](https://github.com/COMP3122-2324-Project/Simpsyber/assets/157147838/b17af0e9-4e59-4bbc-95d1-ccb5171d9a69)

**Comment by Chuenpoly** [comment]
I agree with your point that allowing players to choose staff for different positions at the start of the game or at certain points throughout can provide a more personalized and strategic experience. The game has a leaderboard, meaning that this is a competitive game. If players are provided with a list of the number of vacancies for each role, they will definitely maximise their advantages by selecting all the staff of each role to get higher marks. As a result, everyone will just choose the same combination of staff to play the game.

To make the game more fun, it should restrict the total number of staff but allow different combinations. For example, the total number of staff must be 6. Different players need to use this quota to choose different roles' staff. Player A may choose 1 Designer, 2 software engineers,  2 Infrastructure engineers, and 1 tester. Player B may choose 1 designer, 1 software engineer, 3 Infrastructure engineers, and 1 tester. It makes the game more fun because it also tests the players' human resource allocation ability.

**Comment by jeffreyleett** [comment]
I agree that adding flexibility to the game, which let players select staff for different positions, in order to make it more engaging and dynamic. This kind of personalization will allow players to experiment with various tactics and approaches, not keeping the game from getting repetitive or simple to solve. Finally added complexity and variety will contribute to a more engaging and satisfying gameplay experience.

Not only staff customization, but also adding features to randomize staff abilities or experience. players can face unique challenges and opportunities in team combination and project management. Randomized staff abilities require players to adapt their strategies and make the most suitable team they have. This feature would further enhance replayability and strategic decision-making in the game.

**Comment by LAMYUHIN** [comment]
I think your point about customizing employee positions to add strategy to the game is a good one. As a competitive game with leaderboard, for players to obtain high rankings, they must accurately allocate the best employees to the development work, but the combinations selected under fixed game roles and environments are limited. Therefore, customized positions can bring new variables to the game, thus increasing its fun and longevity.

I suggest more detailed custom allocations be made in the employee ability section. For example, the total number of employee ability points is preset to 20. Players can adjust it through time or money according to their game strategies (employee training); the ability points must be maintained at 20 points to ensure the average character ability value. The picture below shows a simple concept:
![螢幕擷取畫面 2024-03-03 160907](https://github.com/COMP3122-2324-Project/Simpsyber/assets/157567276/fedd6032-06cb-4dc1-bea9-15d1843bef16)
